### PR TITLE
add basic integration of CKEditor w/ Easy Image support

### LINKED ISSUE #1326: Article Post form
Designs coming soon. 

On the front-end, this should behave similar to writing articles on LinkedIn. Users can add any number of content modules, choosing from a short list, e.g. text, image, link, code, etc. In this way, they can build up a long-form post. 

After investigating several prosemirror implementations, we have decided to try to implement this with CKEditor which meets our integration requirements but has more functionality in the standard distribution (similar to django's "batteries included" approach).  

### acceptance criteria:
- [x] #1454 Custom build of ckeditor
- [x] #1455 Light integration of CKEditor
- [x] #1456 add Easy Image support

### related issues
- #1436 backend
- #1549 edit article
- #1457 add embed support

### LINKED ISSUE #1455: Article: Light integration of CKEditor
minimal integration of the editor into the open app

### acceptance criteria
- [x] page template to host CKEditor
- [x] basic text editing 
- [ ] save and load in front-end 
- [ ] read-only rendering

### related issues
- #1436 backend

### LINKED ISSUE #1456: Article: add Easy Image support

### PR DESCRIPTION
**Pull request by aliceriot**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots 
  - [ ] Mobile width screenshots 
  - [ ] tag @ferdi or @pdpinch for review  
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested
- [x] Settings
  - [x] New settings are documented and present in `app.json`
  - [ ] New settings have reasonable development defaults, if applicable

#### What are the relevant tickets?

closes #1456 
closes #1455 
closes #1326 

#### What's this PR do?

This adds a basic integration for CKEditor, including:

- implementing a new `ArticleEditor` component which does the editing and display of saved articles
- an `ARTICLE_UI` feature flag
- adding options and support for editing an article to the post create page
- logic to display an article with the read-only version of the editor on the post detail page
- validation for the post create page
- a backend API for authenticating with CKEditor's easy image service

I think that's about it.

#### How should this be manually tested?

Basically, I think the thing to do is get everything set up and then play around with it, make sure you can save articles with images, view them, etc. To get set up you'll need to get the secrets for CKEditor and add them to `.env`, alongside the `ARTICLE_UI` feature flag.

If you do that you should a new option in the post create page which will allow you to 'write an article'. You should also be able to add images to the article and save it. On the post detail page the article you wrote should be display by the read-only mode of the article editor.

#### Screenshots (if appropriate)

addarticle
addarticleasdfasdfasdfffffval
addarticleasdfasdfasdf
addarticleasdfasdfasdfffff

### CODE REVIEW

**Comment by aliceriot** [comment]
yeah, the picture button appears in the 'paragraph' button which appears to the left when you're editing text. I'm planning to add edit functionality in a follow-up PR.

**Comment by mbertrand** [comment]
Thanks, got it working after setting up a demo EasyImage account.

File: open_discussions/views.py [hunk 1]
```diff
@@ -61,6 +61,8 @@ def index(request, **kwargs):  # pylint: disable=unused-argument
         "recaptchaKey": settings.RECAPTCHA_SITE_KEY,
         "search_page_size": settings.ELASTICSEARCH_DEFAULT_PAGE_SIZE,
         "accepted_social_sites": list(SOCIAL_SITE_NAME_MAP.values()),
+        "article_ui_enabled": features.is_enabled(features.ARTICLE_UI),
```
[the remark this answers is not in the document]
Comment (aliceriot) [reply open_discussions/views.py]: yeah, I think that makes sense, that way we won't show the article UI in the frontend unless everything necessary for it is set-up correctly.

**Comment by aliceriot** [comment]
@mbertrand ok I added that check

File: scripts/test/js_test.sh [hunk 2]
```diff
@@ -11,7 +11,7 @@ elif [[ ! -z "$WATCH" ]]
 then
     export CMD="node ./node_modules/mocha/bin/_mocha --watch"
 else
-    export CMD="node ./node_modules/mocha/bin/_mocha"
+    export CMD="node ./node_modules/mocha/bin/_mocha --exit"
```
Comment (mbertrand) [inline scripts/test/js_test.sh]: JS tests are still timing out for some reason, this didn't seem to fix it?

Comment (aliceriot) [reply scripts/test/js_test.sh]: hm, I was hoping it would because it fixed it locally. I'll look into it and see

Comment (aliceriot) [reply scripts/test/js_test.sh]: ah, I forgot how the tests are called on travis. I think it should be good now (we'll see I suppose).

Comment (aliceriot) [reply scripts/test/js_test.sh]: ok looks like that fixed the test exit issue! build is still failing b/c of the sass linter though, oops

Comment (aliceriot) [reply scripts/test/js_test.sh]: haha, fixed the sass linter, ran into a thing with uglify...

**Comment by aliceriot** [comment]
@mbertrand ok I think the build issue is FINALLY sorted out. I ended up having to upgrade us to webpack v4. The issue was that the build setup that they distribute for ckeditor, which we're using in our custom build, is using webpack 4, and for some reason our existing webpack 3 setup was not happy with incorporating that build artifact into our build. It was throwing an error when it tried to run the uglify.js minifier on the whole bundled output, because I think the webpack4 build was already minifying the code (but in an incompatible way somehow? not sure).

anyhow, I upgraded everything to webpack4 here and dealt with the configuration changes necessary to make that work. I think it should all be good, but it would be good to test somewhat extensively to make sure I didn't miss anything! we should try it out on the PR build if possible, or we could just wait to test on RC, because I want to make absolutely sure the build is working in a prod environment.

**Comment by mbertrand** [comment]
@aliceriot thanks for the update, I will test it out.

**Comment by mbertrand** [comment]
@aliceriot testing locally, everything still seems to work on the front end, still some test failures though.

**Comment by aliceriot** [comment]
ok I think I got that python test fixed, I think things should be good now
